runners/local-artifact-service-java/src/main/java/org/apache/beam/artifact/local/LocalFileSystemArtifactRetrievalService.java

```diff
@@ -27,76 +27,78 @@
 import java.io.IOException;
 import java.nio.ByteBuffer;
 import java.nio.channels.FileChannel;
-import java.util.ArrayList;
-import java.util.List;
+import java.nio.channels.FileChannel.MapMode;
 import org.apache.beam.model.jobmanagement.v1.ArtifactApi;
+import org.apache.beam.model.jobmanagement.v1.ArtifactApi.ArtifactChunk;
 import org.apache.beam.model.jobmanagement.v1.ArtifactApi.GetManifestResponse;
+import org.apache.beam.model.jobmanagement.v1.ArtifactApi.Manifest;
 import org.apache.beam.model.jobmanagement.v1.ArtifactRetrievalServiceGrpc;
 import org.apache.beam.runners.fnexecution.artifact.ArtifactRetrievalService;
 
 /** An {@code ArtifactRetrievalService} which stages files to a local temp directory. */
 public class LocalFileSystemArtifactRetrievalService
     extends ArtifactRetrievalServiceGrpc.ArtifactRetrievalServiceImplBase
     implements ArtifactRetrievalService {
-  private static final long DEFAULT_CHUNK_SIZE = 2 * 1024 * 1024;
+  private static final int DEFAULT_CHUNK_SIZE = 2 * 1024 * 1024;
 
   public static LocalFileSystemArtifactRetrievalService forRootDirectory(File base) {
     return new LocalFileSystemArtifactRetrievalService(base);
   }
 
   private final LocalArtifactStagingLocation location;
+  private final Manifest manifest;
 
   private LocalFileSystemArtifactRetrievalService(File rootDirectory) {
     this.location = LocalArtifactStagingLocation.forExistingDirectory(rootDirectory);
-  }
-
-  /**
-   * Get the manifest of artifacts that can be retrieved by this {@link
-   * LocalFileSystemArtifactRetrievalService}.
-   */
-  private ArtifactApi.Manifest getManifest() throws IOException {
-    return ArtifactApi.Manifest.parseFrom(new FileInputStream(location.getManifestFile()));
+    try {
+      this.manifest =
+          ArtifactApi.Manifest.parseFrom(new FileInputStream(location.getManifestFile()));
```

Comment:
You need to close the FileInputStream otherwise you'll leak file handles.

Reply:
Fixed.